CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatMain.lua

```diff
@@ -0,0 +1,739 @@
+			channelObj:AddMessageLabelToLog(messageObject)
+		end
+	end
+
+	return processedCommand
+end
+
+--// Event for making player say chat message.
+ChatBar:GetTextBox().FocusLost:connect(function(enterPressed, inputObject)
+	if (enterPressed) then
+		local message = string.sub(ChatBar:GetTextBox().Text, 1, 300) -- max of something whenever
+		ChatBar:GetTextBox().Text = ""
+		
+		if (message ~= "" and not ProcessChatCommands(message)) then
+			message = string.gsub(message, "\n", "")
+			message = string.gsub(message, "[ ]+", " ")
+			
+			local currentChannel = ChatWindow:GetCurrentChannel()
+			if (currentChannel) then
+				MessageSender:SendMessage(message, currentChannel.Name)
+
+				if (currentChannel.Name == "All") then
+					--// Sends signal to eventually call Player:Chat() to handle C++ side legacy stuff.
+					moduleApiTable.MessagePosted:fire(message) 
+				end
+			else
+				MessageSender:SendMessage(message, nil)
+				
+			end
+		end
+		
+	end
+end)
+
+EventFolder.OnNewMessage.OnClientEvent:connect(function(fromSpeaker, channel, message)
+	local channelObj = ChatWindow:GetChannel(channel)
+	if (channelObj) then
+		local messageObject = MessageLabelCreator:CreateMessageLabel(fromSpeaker, message)
+		channelObj:AddMessageLabelToLog(messageObject)
+		
+		ChannelsBar:UpdateMessagePostedInChannel(channel)
+		
+		moduleApiTable.MessageCount = moduleApiTable.MessageCount + 1
+		moduleApiTable.MessagesChanged:fire(moduleApiTable.MessageCount)
+
+		DoFadeInFromNewInformation()
+
+	else
+		warn("Just received chat message for channel I'm not in [" .. channel .. "]")
+		
+	end
+end)
+
+EventFolder.OnNewSystemMessage.OnClientEvent:connect(function(message, channel)
+	channel = channel or "System"
+	
+	local channelObj = ChatWindow:GetChannel(channel)
+	if (channelObj) then
+		local messageObject = MessageLabelCreator:CreateSystemMessageLabel(message)
+		channelObj:AddMessageLabelToLog(messageObject)
+		
+		ChannelsBar:UpdateMessagePostedInChannel(channel)
+		
+		moduleApiTable.MessageCount = moduleApiTable.MessageCount + 1
+		moduleApiTable.MessagesChanged:fire(moduleApiTable.MessageCount)
+
+		DoFadeInFromNewInformation()
+
+	else
+		warn("Just received system message for channel I'm not in [" .. channel .. "]")
+		
+	end
+end)
+
+EventFolder.OnChannelJoined.OnClientEvent:connect(function(channel, welcomeMessage)
+	local channelObj = ChatWindow:AddChannel(channel)
+
+	if (channelObj) then
+		if (channel == "All") then
+			ChatWindow:SwitchCurrentChannel(channel)
+		end
+		
+		if (welcomeMessage ~= "") then
+			local messageObject = MessageLabelCreator:CreateWelcomeMessageLabel(welcomeMessage)
+			channelObj:AddMessageLabelToLog(messageObject)
+		end
+
+		DoFadeInFromNewInformation()
+	end	
+	
+end)
+
+EventFolder.OnChannelLeft.OnClientEvent:connect(function(channel)
+	ChatWindow:RemoveChannel(channel)
+
+	DoFadeInFromNewInformation()
+end)
+
+EventFolder.OnMuted.OnClientEvent:connect(function(channel)
+	-- handle
```

Comment:
It ended up being unused. Before we got art from the art team and the chat window UI looked bad, this feature was used to take away the chat bar when a player was muted in a channel. What was found out was that it was very inconvenient when a player couldn't chat out commands like "/join channel" when they were muted in every channel they were in. When recreating the window UI script after getting the art, this part was added in but nothing was done with it.

I'll update the comment to be more clear about this in case in the future something will want to be done with it.